File: go/apis/slo/v1/slo_mapping.go
package v1

// +kubebuilder:validation:Required

// SloMapping can be used as a base type for specific SLO mapping types by embedding it inline.
// It includes the properties required by most SLO mappings.
type SloMapping struct {
	// Specifies the target on which to execute the elasticity strategy.
	TargetRef SloTarget `json:"targetRef"`

	// Specifies the type of ElasticityStrategy to use for this SLO mapping.
	ElasticityStrategy ElasticityStrategyKind `json:"elasticityStrategy"`

	// Configures the duration of the period after the last elasticity strategy execution,
	// during which the strategy will not be executed again (to avoid unnecessary scaling).
	//
	// +optional
	StabilizationWindow *StabilizationWindow `json:"stabilizationWindow,omitempty"`

	// Any static configuration parameters, which are unknown to the SLO, but which may be required to configure
	// the chosen kind of elasticity strategy should be specified here. They will be copied over
	// into the spec of the elasticity strategy.
	//
	// // Note that the Go deepCopy() method will only be able to create shallow copies of this field,
	// because it does not know its internal structure.
	// +optional
	StaticElasticityStrategyConfig *ArbitraryConfigObject `json:"staticElasticityStrategyConfig,omitempty"`
}

File: ts/apps/slo/cpu-usage-slo-controller/src/app/model/sloc-crds/slo-mapping.ts
import { V1CrossVersionObjectReference } from '@kubernetes/client-node';
import { ElasticityStrategyKind } from './elasticity-strategy-kind';

/**
 * Base interface for SloMappings.
 */
export interface SloMapping {

    /** Specifies the target on which to execute the elasticity strategy. */
    targetRef: V1CrossVersionObjectReference;

    /** Specifies the type of ElasticityStrategy to use for this SLO mapping. */
    elasticityStrategy: ElasticityStrategyKind;

}
